tools/SetupFlow/DevHome.SetupFlow/ViewModels/AddRepoViewModel.cs

```diff
@@ -298,12 +328,127 @@ private void CancelButtonPressed()
+        }
+    }
+
+    /// <summary>
+    /// Makes the MenuFlyout object used to display multple accounts in the repo tool.
+    /// </summary>
+    /// <returns>The MenuFlyout to display.</returns>
+    /// <remarks>
+    /// The MenuFlyout is constructed in code to apply the correct callback methods on the Menu Items.
+    /// </remarks>
+    private MenuFlyout ConstructFlyout()
+    {
+        AccountsToShow = new MenuFlyout();
+        var newMenu = new MenuFlyout();
+        foreach (var account in Accounts)
+        {
+            var thisMenuItem = new MenuFlyoutItem();
+            thisMenuItem.Name = account;
+            thisMenuItem.Text = account;
+            thisMenuItem.Click += MenuItemClick;
+            newMenu.Items.Add(thisMenuItem);
+        }
+
+        newMenu.Items.Add(new MenuFlyoutSeparator());
+        var thatMenuItem = new MenuFlyoutItem();
+        thatMenuItem.Text = _stringResource.GetLocalized("RepoToolAddAnotherAccount");
+        thatMenuItem.Click += AddAccountClicked;
+        newMenu.Items.Add(thatMenuItem);
+
+        return newMenu;
+    }
+
+    /// <summary>
+    /// The bottom of the MenuFlyout has a button to log into another account.  Handle logging the user in.
+    /// </summary>
+    /// <param name="sender">The object that sent this event.</param>
+    /// <param name="e">The arguments</param>
+    /// <remarks>
+    /// This calls MenuItemClick to poulate the list of repos if a new account is detected.
+    /// </remarks>
+    private async void AddAccountClicked(object sender, RoutedEventArgs e)
+    {
+        if (sender is MenuFlyoutItem selectedItem &&
+            selectedItem.Text.Equals(_stringResource.GetLocalized("RepoToolAddAnotherAccount"), StringComparison.OrdinalIgnoreCase))
+        {
+            // If the user selects repos from account 1, then logs into account 2 and does not save between those two actions
+            // _previouslySelectedRepos will be empty.  The result is the repos in account 1 will not be selected if the user navigates
+            // to account 1 after logging into account 2.
+            // Save the repos here in that case.
+            if (!_previouslySelectedRepos.Any())
+            {
+                _previouslySelectedRepos.AddRange(EverythingToClone);
+            }
+
+            ShowRepoPage = Visibility.Collapsed;
+
+            // Store the logged in accounts to help figure out what account the user logged into.
+            var loggedInAccounts = await Task.Run(() => _providers.GetAllLoggedInAccounts(_selectedRepoProvider));
+            await LogUserIn(_selectedRepoProvider, _addRepoDialog.GetLoginUiContent(), true);
+            var loggedInAccountsWithNewAccount = await Task.Run(() => _providers.GetAllLoggedInAccounts(_selectedRepoProvider));
+
+            ShowRepoPage = Visibility.Visible;
+            Accounts = new ObservableCollection<string>(loggedInAccountsWithNewAccount.Select(x => x.LoginId));

```

Comment:
Check for x.Url which is unique (since GHES breaks uniqueness of LoginId)

Reply:
I'm going to leave this as is for now.  Changing everything to support .Url is too much for this PR.